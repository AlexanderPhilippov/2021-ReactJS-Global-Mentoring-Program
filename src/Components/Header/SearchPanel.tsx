import image from 'Assets/Images/movies-tiles.jpg'
import React from 'react'
import HeaderTextLogo from './HeaderTextLogo'
import { SearchPanelProps } from './models'
import './styles.scss'

const SearchPanel: React.FC<SearchPanelProps> = ({ modalToggle }) => {
    return (
        <>
            <img className="header__image" src={image} />
            <HeaderTextLogo />
            <input
                className="header__add-movie_button"
                type="button"
                value="+ add movie"
                onClick={() => modalToggle((prevState) => !prevState)}
            />
            <div className="header__search">
                <label htmlFor="searchMovieInput">Find your movie</label>
                <input
                    id="searchMovieInput"
                    type="text"
                    placeholder="What do you want to watch?"
                />
                <input type="button" value="search" />
            </div>
        </>
    )
}

export default SearchPanel
