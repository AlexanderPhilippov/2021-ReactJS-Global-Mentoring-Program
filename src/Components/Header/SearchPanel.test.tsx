/**
 * @jest-environment jsdom
 */
import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import SearchPanel from './SearchPanel'
import { MemoryRouter } from 'react-router-dom'

const mockStore = configureStore()
it('MovieDetailsPanel', () => {
    const store = mockStore({})
    const { asFragment } = render(
        <Provider store={store}>
            <MemoryRouter>
                <SearchPanel />
            </MemoryRouter>
        </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
})
