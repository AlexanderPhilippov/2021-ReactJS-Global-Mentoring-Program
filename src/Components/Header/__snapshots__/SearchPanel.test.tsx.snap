// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPanel 1`] = `
<DocumentFragment>
  <img
    class="header__image"
    src=""
  />
  <div
    class="header__text-logo"
  >
    <span>
      netflix
    </span>
    roulette
  </div>
  <input
    class="header__add-movie_button"
    type="button"
    value="+ add movie"
  />
  <div
    class="header__search"
  >
    <label
      for="searchMovieInput"
    >
      Find your movie
    </label>
    <input
      id="searchMovieInput"
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <input
      type="button"
      value="search"
    />
  </div>
</DocumentFragment>
`;
